UPtables/app/controllers/ects_modules_controller.rb

```diff
@@ -0,0 +1,77 @@
+class EctsModulesController < ApplicationController
+  before_action :set_ects_module, only: [:show, :edit, :update, :destroy]
+
+  # GET /ects_modules
+  # GET /ects_modules.json
+  def index
+    @ects_modules = EctsModule.all
+  end
+
+  # GET /ects_modules/1
+  # GET /ects_modules/1.json
+  def show
+  end
+
+  # GET /ects_modules/new
+  def new
+    @ects_module = EctsModule.new
+  end
+
+  # GET /ects_modules/1/edit
+  def edit
+  end
+
+  # POST /ects_modules
+  # POST /ects_modules.json
+  def create
+    @ects_module = EctsModule.new(ects_module_params)
+    @ects_module.course_ids = params[:ects_module][:course_ids]
+
+    respond_to do |format|
+      if @ects_module.save
+        format.html { redirect_to @ects_module, notice: 'ects module was successfully created.' }
+        format.json { render :show, status: :created, location: @ects_module }
+      else
+        format.html { render :new }
+        format.json { render json: @ects_module.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /ects_modules/1
+  # PATCH/PUT /ects_modules/1.json
+  def update
+    @ects_module.course_ids = params[:ects_module][:course_ids]
```

Comment:
[solution](http://stackoverflow.com/questions/16549382/how-to-permit-an-array-with-strong-parameters)

I'm working on it. You will get a nice fixed controller spec.